### PR TITLE
WIP : Adding test-OCS347

### PR DESCRIPTION
**Pull request by parikshithb**
Will make few changes when CSI to flex is merged also waiting for default templates.

================================================================================================  summary =================================================================================================
tests/test_rbd_csi_default_sc.py::TestCaseOCS347::test_347
tests/test_rbd_csi_default_sc.py::TestCaseOCS347::test_347

-- Docs: https://docs.pytest.org/en/latest/warnings.html
====================================================================================== 1 passed in 44.00 seconds ======================================================================================

### CODE REVIEW

File: templates/ocs-deployment/pool.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: {{ pool_name | default ('rbd') }}
+  namespace: rook-ceph
```
Comment (RazTamir) [inline templates/ocs-deployment/pool.yaml]: Please add an option to pass parameter here.
Unless it is not allowed, ignore this comment

File: tests/test_rbd_csi_default_sc.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    Create Blockpool with default values
+    """
+    file_y = templating.generate_yaml_from_jinja2_template_with_data(
+        RBD_POOL_YAML
+    )
+    with open(TEMP_YAML_FILE, 'w') as yaml_file:
+        yaml.dump(file_y, yaml_file, default_flow_style=False)
+    log.info(f"Creating a new CephBlockPool with default name")
+    assert POOL.create(yaml_file=TEMP_YAML_FILE)
+    sleep(5)
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: please don't use sleep. never (:
Everything that needs waiting, we will sample to check the status

File: tests/test_rbd_csi_default_sc.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    """
+    file_y = templating.generate_yaml_from_jinja2_template_with_data(
+        SC_RBD_YAML
+    )
+    with open(TEMP_YAML_FILE, 'w') as yaml_file:
+        yaml.dump(file_y, yaml_file, default_flow_style=False)
+    log.info(f"Creating a RBD StorageClass with default values")
+    assert SC.create(yaml_file=TEMP_YAML_FILE)
+    log.info(f"Sleeping for 5 sec to let StorageClass get created")
+    sleep(2)
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: same

File: tests/test_rbd_csi_default_sc.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+def validate_pool_creation(pool_name):
+    """
+    Check whether default blockpool is created or not at ceph and as well
+    OCS side
+
+    :param pool_name:
+    :return:
+    """
+    ceph_validate = False
+    OCS_validate = False
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: ocs_validate

File: tests/test_rbd_csi_default_sc.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    for item in sample:
+        if item.metadata.name == pool_name:
+            OCS_validate = True
+    if ceph_validate and OCS_validate:
+        log.info("Pool got created successfully from Ceph and OCS side")
+        return True
+    else:
+        return False
+
+def get_toolbox_pod_handler():
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: No need as we already have this https://github.com/red-hat-storage/ocs-ci/blob/master/ocs/ocp.py#L252

File: tests/test_rbd_csi_default_sc.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    OCS side
+
+    :param pool_name:
+    :return:
+    """
+    ceph_validate = False
+    OCS_validate = False
+    cmd = "ceph osd lspools|grep " + str(pool_name) + " |awk '{print$2}'"
+    took_box = get_toolbox_pod_handler()
+    out, err, ret = took_box.exec_command(cmd=cmd, timeout=20)
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: please use https://github.com/red-hat-storage/ocs-ci/blob/master/ocs/ocp.py#L267

File: tests/test_rbd_csi_default_sc.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    Check whether default blockpool is created or not at ceph and as well
+    OCS side
+
+    :param pool_name:
+    :return:
+    """
+    ceph_validate = False
+    OCS_validate = False
+    cmd = "ceph osd lspools|grep " + str(pool_name) + " |awk '{print$2}'"
+    took_box = get_toolbox_pod_handler()
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: please use https://github.com/red-hat-storage/ocs-ci/blob/master/ocs/ocp.py#L252

File: tests/test_rbd_csi_default_sc.py [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+        if item.metadata.name in sc_name:
+            log.info(f"StorageClass got created successfully")
+            return True
+    return False
+
+def create_secret_rbd(**kwargs):
+    """
+    This will create Secret file which will be used for creating StorageClass
+
+    :return:
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: Please update the passed arguments according to the documentation

File: tests/test_rbd_csi_default_sc.py [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+    return True
+
+def validate_pvc(pvc_name):
+    """
+    This will check if PVC created is bound successfully or not
+
+    :param pvc_name:
+    :return:
+    """
+    pvc_list = PVC.get(resource_name=pvc_name)
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: when you provide a resource_name you can expect to get only one item and not a list

File: tests/test_rbd_csi_default_sc.py [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+    def test_347(self):
+
+        assert create_rbd_pool()
+        assert validate_pool_creation("rbd")
+        assert create_storageclass_rbd()
+        assert validate_storageclass("ocsci-csi-rbd")
+        get_client_admin_keyring()
+        assert create_pvc()
+        assert validate_pvc("rbd-pvc")
+        return StatusOfTest.PASSED
```
Comment (RazTamir) [inline tests/test_rbd_csi_default_sc.py]: Add empty line at the end of the file.
pep8 requires an empty line

File: tests/test_rbd_csi_default_sc.py [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+    kind='Deployment', namespace="rook-ceph"
+)
+SECRET = ocp.OCP(
+    kind='Secret', namespace="default"
+)
+PVC = ocp.OCP(
+    kind='PersistentVolumeClaim', namespace="rook-ceph"
+)
+
+def create_rbd_pool():
```
Comment (vasukulkarni) [inline tests/test_rbd_csi_default_sc.py]: This should be a OCS function call, I am not going to hold this one but will remodify this later after the cluster context is fixed.

**Comment by vasukulkarni** [comment]
we should move the yamls' in templates/rbd/

File: tests/test_rbd_csi_default_sc.py [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+    log.info(f"Sleeping for 5 sec to let StorageClass get created")
+    sleep(2)
+    return True
+
+def validate_pool_creation(pool_name):
+    """
+    Check whether default blockpool is created or not at ceph and as well
+    OCS side
+
+    :param pool_name:
```
Comment (ebenahar) [inline tests/test_rbd_csi_default_sc.py]: Use google sphynx docstring style:
https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000218290-Configure-google-docstring

File: tests/test_rbd_csi_default_sc.py (hunk 10 above)
Comment (dahorak) [reply tests/test_rbd_csi_default_sc.py]: Please ensure, that each file is ending with new line (also `pool.yaml` and `storageclass-csi-rbd.yaml`).

**Comment by parikshithb** [comment]
Closing this PR. Will raise new PR with latest changes.

**Pull request closed without merging**